Code/GraphMol/FileParsers/FileParserUtils.h

```diff
@@ -32,26 +33,133 @@ T stripSpacesAndCast(const std::string &input, bool acceptSpaces = false) {
     return boost::lexical_cast<T>(trimmed);
   }
 }
-RDKIT_FILEPARSERS_EXPORT int toInt(const std::string &input, bool acceptSpaces = false);
-RDKIT_FILEPARSERS_EXPORT double toDouble(const std::string &input, bool acceptSpaces = true);
+RDKIT_FILEPARSERS_EXPORT int toInt(const std::string &input,
+                                   bool acceptSpaces = false);
+RDKIT_FILEPARSERS_EXPORT double toDouble(const std::string &input,
+                                         bool acceptSpaces = true);
 
 // reads a line from an MDL v3K CTAB
-RDKIT_FILEPARSERS_EXPORT std::string getV3000Line(std::istream *inStream, unsigned int &line);
+RDKIT_FILEPARSERS_EXPORT std::string getV3000Line(std::istream *inStream,
+                                                  unsigned int &line);
 
 // nAtoms and nBonds are ignored on input, set on output
-RDKIT_FILEPARSERS_EXPORT bool ParseV3000CTAB(std::istream *inStream, unsigned int &line, RWMol *mol,
-                    Conformer *&conf, bool &chiralityPossible,
-                    unsigned int &nAtoms, unsigned int &nBonds,
-                    bool strictParsing = true, bool expectMEND = true);
+RDKIT_FILEPARSERS_EXPORT bool ParseV3000CTAB(
+    std::istream *inStream, unsigned int &line, RWMol *mol, Conformer *&conf,
+    bool &chiralityPossible, unsigned int &nAtoms, unsigned int &nBonds,
+    bool strictParsing = true, bool expectMEND = true);
 
 // nAtoms and nBonds are used
-RDKIT_FILEPARSERS_EXPORT bool ParseV2000CTAB(std::istream *inStream, unsigned int &line, RWMol *mol,
-                    Conformer *&conf, bool &chiralityPossible,
-                    unsigned int &nAtoms, unsigned int &nBonds,
-                    bool strictParsing = true);
+RDKIT_FILEPARSERS_EXPORT bool ParseV2000CTAB(
+    std::istream *inStream, unsigned int &line, RWMol *mol, Conformer *&conf,
+    bool &chiralityPossible, unsigned int &nAtoms, unsigned int &nBonds,
+    bool strictParsing = true);
 
 RDKIT_FILEPARSERS_EXPORT Atom *replaceAtomWithQueryAtom(RWMol *mol, Atom *atom);
+
+template <typename T>
+void applyMolListPropsToAtoms(ROMol &mol, const std::string &prefix,
+                              const std::string &missingValueMarker = "n/a") {
+  for (auto pn : mol.getPropList()) {
+    if (pn.find(prefix) == 0 && pn.size() > prefix.size()) {
+      std::string strVect = mol.getProp<std::string>(pn);
+      std::vector<std::string> tokens;
+      boost::split(tokens, strVect, boost::is_any_of(" \t\n"),
+                   boost::token_compress_on);
+      if (tokens.size() < mol.getNumAtoms()) {
+        BOOST_LOG(rdWarningLog)
+            << "Property list " << pn << " too short, only " << tokens.size()
+            << " elements found. Ignoring it." << std::endl;
+        continue;
+      }
+      std::string mv = missingValueMarker;
```

Comment:
Isn't the standard missingValueMarker n/a?

So with this code, one can't write:  [n/a] 1 2 since the token can only be one character.  Also, what if the form is:

[] 1 2 

This would make the missing token "]"

I would take everything between [ and ] for consistency.

Reply:
I was incorrect about what substr was doing.

Reply:
Add regression test for atom prop with multiple chars

```[n/a] 1 2 [n/a]```